tools/ports/benchmark/portfile.cmake

```diff
@@ -0,0 +1,49 @@
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+  message(FATAL_ERROR "${PORT} does not currently support UWP")
+endif()
+
+if (VCPKG_PLATFORM_TOOLSET STREQUAL "v140")
+  # set(CMAKE_C_COMPILER_WORKS 1)
+  # set(CMAKE_CXX_COMPILER_WORKS 1)
+  set(CMAKE_C_COMPILER cl.exe)
+  set(CMAKE_CXX_COMPILER cl.exe)
+  set(MSVC_TOOLSET_VERSION 140)
+  # set(VCPKG_VISUAL_STUDIO_PATH "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0")
+  # set(VCPKG_PLATFORM_TOOLSET v140)
+else()
+  # Make sure vs2019 compiled binaries are compat with vs2017

```

Comment:
This was needed to pin to specific commit ID. Also the issue in the mainline `vcpkg` that by default, if built with vs2019 - it installs the library globally and that library cannot be reused by vs2017 installation on the same machine. That is because ABI compatibility is INSUFFICIENT on Windows: the libraries must all be linked by the lowest version of the toolchain. The flag `/d2FH4-` allows to turn off vs2019-specific exception semantics and make the library link successfully with vs2017-compiled binaries, even if Google Benchmark itself is compiled with vs2019.

Reply:
Is sharing between VS 2019 and 2017 the reason to add this vcpkg port? I think vcpkg includes port of `benchmark`.